pkg/framework/wait.go

```diff
@@ -11,7 +11,7 @@ const (
 	defaultTimeout = 1800
 )
 
-func waitFor(f func() error) error {
+func WaitFor(f func() error) error {
```

Comment:
Made this public to reuse it in the new package.

Reply:
I am for dropping this and just using the backoff thing. Like why do we have to implement this again and use a different thing that does the same? It is harder to configure and when checking the logs of e2e tests its behaviour is then different. Don't understand that.

Reply:
Done, this will need cleaning as well - we have to `pkg/framework/wait.go` file, `pkg/framework/backoff.go`, `pkg/framework/retry.go` and `pkg/wait/wait.go`.